client/src/components/Search.tsx

```diff
@@ -1,27 +1,49 @@
 import React, { useState, useRef, useEffect } from "react";
 import { SearchNodeSuggestion } from "../types/roomsearch/SearchNodeSuggestion";
+import { useRouter } from "next/router";
+import { RoomSearch } from "../logic/interfaces/RoomSearch";
 
 type Prop = {
-  roomSearcher: (searchedText: string) => SearchNodeSuggestion[];
+  roomSearcher: RoomSearch;
```

Comment:
Look at this change. Before, you should just pass different function for start and destination search and this component didn't need to know about how search form works and where it is used.

Now, you can only call `roomSearcher.sortedSuggestionsForStart` instead of passed `roomSearcher` function.

I think you changed this because you wanted to use `findRoomByNodeId` function. You can implement that by adding another function prop for that: `findRoomByNodeId: (nodeId: string): SearchNodeSuggestion`.